button/internal/button.ts

```diff
@@ -164,4 +164,23 @@ export abstract class Button extends LitElement implements FormSubmitter {
   private handleSlotChange() {
     this.hasIcon = this.assignedIcons.length > 0;
   }
+  
+  /**
+   * Whether the element should be disabled either through its own `disabled` or
+   * its formDisabled.
+   */
+  private get shouldBeDisabled() {
+    return this.disabled || this.formDisabled;
+  }
+  
+  /**
+   * Whether the element is currently disabled due to form state.
+   */
+  private formDisabled = false;
+
+  /** @private */
+  formDisabledCallback(formDisabled: boolean) {
+    this.formDisabled = formDisabled;
```

Comment:
I planned to do that but it would be a wrong behavior. If you set `disabled` of the control at `formDisabledCallback`, the control itself would be disabled.

So in case you have a `fieldset` with two controls, one is disabled (by itself) and one is not, when the fieldset `disabled` is lifted, one control should still be disabled:

```html
<fieldset>
   <input disabled /> <!-- This should stay disabled if the fieldset is disabled and then enabled -->
   <button></button>
</fieldset>
```

Reply:
Gotcha, thanks for the explanation!